resources/articles/s2s-auth.md

```diff
@@ -0,0 +1,121 @@
+:title Service-to-service auth
+
+:frontpage
+:category api-integration
+:aside
+## Table of Contents
+
+<spid-toc></spid-toc>
+
+## Prerequisites
+
+In order to make use of this information, you need to know your:
+
+- client ID
+- client secret
+
+## See also
+
+- [Obtaining tokens](/oauth/token/)
+- [Token introspection](/token-introspection/)
+
+:body
+
+In addition to the user auth facilities provided by the platform, SPiD also provides a mechanism for
+when two services need to communicate directly without a user present. In these situations, a service
+needs to authenticate and authorize the other services which access its protected resources. We refer
+to this as service-to-service (henceforth S2S) auth.
+
+S2S auth in SPiD is made up of three basic steps:
+1. A client [obtains a client (a.k.a. server) token](/oauth/token/) via
+a `client_credentials` grant.
+1. The client makes a request to the resource server passing the newly acquired token as a
+[bearer token](https://tools.ietf.org/html/rfc6750).
+1. The resource server [introspects](/token-introspection/) the token it receives and validates the properties
+in the introspected token. If it is valid, the request is allowed to proceed. If it is not, the
+request is rejected.
+
+![Server-to-server communication flow](/images/server-to-server-comm.png)
+
+The following sections break down these high-level steps in more detail.
+
+## Obtaining a token
+
+Before making a call to a resource server, the calling client first needs to [obtain a token](/oauth/token/).
+For this, a `client_credentials` grant is made using the caller's client ID and secret. Additionally,
+the caller should specify the resource (a.k.a. audience) trying to be accessed via the `resource`
```

Comment:
Relation between resource and audience is not 1:1, so maybe remove "(a.k.a. audience)".

Reply:
Fixed